We have some conflicts. 

Initially lint-test was showing this issue.
```
LibMambaUnsatisfiableError: Encountered problems while solving:
  - package pydantic-settings-2.12.0-py310h06a4308_0 requires pydantic >=2.7.0, but none of the providers can be installed

Could not solve for environment specs
The following packages are incompatible
├─ anaconda-auth =* * is not installable because it requires
│  └─ anaconda-cli-base [>=0.5.2 *|>=0.5.4 *|>=0.7.0 *] with the potential options
│     ├─ anaconda-cli-base [0.5.2|0.5.3|0.5.4|0.6.0] would require
│     │  └─ pydantic-settings >=2.3 *, which requires
│     │     └─ pydantic >=2.7.0 *, which can be installed;
│     └─ anaconda-cli-base 0.7.0 would require
│        ├─ anaconda-auth >=0.12.0 *, which can be installed;
│        └─ pydantic-settings >=2.3 *, which can be installed (as previously explained);
└─ pydantic =1.10 * is not installable because it conflicts with any installable versions previously reported.
```

Updating pydantic to the latest version then resulted in this issue.

```
LibMambaUnsatisfiableError: Encountered problems while solving:
  - package fastapi-0.94.0-pyhd8ed1ab_0 requires pydantic >=1.6.2,!=1.7,!=1.7.1,!=1.7.2,!=1.7.3,!=1.8,!=1.8.1,<2.0.0, but none of the providers can be installed

Could not solve for environment specs
The following packages are incompatible
├─ fastapi =0.94 * is installable and it requires
│  └─ pydantic >=1.6.2,(!=1.7,(!=1.7.1,(!=1.7.2,(!=1.7.3,(!=1.8,(!=1.8.1,<2.0.0)))))) *, which can be installed;
└─ pydantic =2.12 * is not installable because it conflicts with any installable versions previously reported.
```

Updating fastapi to the latest version removed all the conflicts in the dependencies, but results in errors in the fast API tests, presumably due to changes in functionality.

```
==================================== ERRORS ====================================
_________________ ERROR collecting fastapi_tests/test_main.py __________________
ImportError while importing test module '/home/runner/work/AtLAST_sensitivity_calculator/AtLAST_sensitivity_calculator/fastapi_tests/test_main.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/share/miniconda/lib/python3.10/site-packages/_pytest/python.py:618: in _importtestmodule
    mod = import_path(self.path, mode=importmode, root=self.config.rootpath)
/usr/share/miniconda/lib/python3.10/site-packages/_pytest/pathlib.py:533: in import_path
    importlib.import_module(module_name)
/usr/share/miniconda/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:688: in _load_unlocked
    ???
/usr/share/miniconda/lib/python3.10/site-packages/_pytest/assertion/rewrite.py:168: in exec_module
    exec(co, module.__dict__)
fastapi_tests/test_main.py:2: in <module>
    from fastapi.testclient import TestClient
/usr/share/miniconda/lib/python3.10/site-packages/fastapi/__init__.py:7: in <module>
    from .applications import FastAPI as FastAPI
/usr/share/miniconda/lib/python3.10/site-packages/fastapi/applications.py:13: in <module>
    from fastapi import routing
/usr/share/miniconda/lib/python3.10/site-packages/fastapi/routing.py:24: in <module>
    from fastapi import params
/usr/share/miniconda/lib/python3.10/site-packages/fastapi/params.py:8: in <module>
    from fastapi.openapi.models import Example
/usr/share/miniconda/lib/python3.10/site-packages/fastapi/openapi/models.py:5: in <module>
    from fastapi._compat import with_info_plain_validator_function
/usr/share/miniconda/lib/python3.10/site-packages/fastapi/_compat/__init__.py:16: in <module>
    from .v2 import BaseConfig as BaseConfig
/usr/share/miniconda/lib/python3.10/site-packages/fastapi/_compat/v2.py:18: in <module>
    from pydantic import BaseModel, ConfigDict, Field, TypeAdapter, create_model
E   ImportError: cannot import name 'TypeAdapter' from 'pydantic' (/usr/share/miniconda/lib/python3.10/site-packages/pydantic/__init__.cpython-310-x86_64-linux-gnu.so)
=========================== short test summary info ============================
ERROR fastapi_tests/test_main.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 0.30s ===============================
Error: Process completed with exit code 2.
```

--- a/atlast_sc/models.py
+++ b/atlast_sc/models.py
@@ -53,7 +53,7 @@ class ValueWithUnits(BaseModel):
     value: Union[float, Quantity]
     unit: Union[str, None]
 
-    @root_validator
+    @root_validator(skip_on_failure=True)
     @classmethod
     def validate_fields(cls, field_values):
         """

--- a/atlast_sc/models.py
+++ b/atlast_sc/models.py
@@ -118,7 +118,7 @@ class UserInput(BaseModel):
         ValueWithUnits(value=Data.elevation.default_value,
                        unit=Data.elevation.default_unit)
 
-    @root_validator
+    @root_validator(skip_on_failure=True)
     @classmethod
     def validate_t_int_or_sens_initialised(cls, field_values):
         """
@@ -172,7 +172,7 @@ class CalculationInput(BaseModel):
         ValueWithUnits(value=Data.t_cmb.default_value,
                        unit=Data.t_cmb.default_unit)
 
-    @root_validator
+    @root_validator(skip_on_failure=True)
     @classmethod
     def validate_fields(cls, field_values):
         """

--- a/docs/source/requirements.txt
+++ b/docs/source/requirements.txt
@@ -1,5 +1,5 @@
 sphinx-rtd-theme>=2.0.0
-nbsphinx==0.9
+nbsphinx==0.9.8
 sphinx-copybutton
 sphinxcontrib.plantuml==0.25
 myst-parser==2.0.0

--- a/environment.yml
+++ b/environment.yml
@@ -15,6 +15,7 @@ dependencies:
   - nbsphinx==0.9
   - python-dotenv=0.21
   - pydantic=2.12
+  - pydantic
   - pandoc
   - build
   - flake8

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -23,7 +23,7 @@ dependencies = [
     "numpy == 1.26.*",
     "scipy == 1.11.*",
     "pyyaml == 6.0.*",
-    "pydantic == 1.10.*"
+    "pydantic == 2.12.*"
 ]
 
 [project.urls]

